### PR TITLE
#1794 Display functional data

### PR DESCRIPTION
**Pull request by h-tong**
**Steps to test #1794**

1. Add Variant by CA ID
2. In the evidence view, click the **Experimental** tab
3. In the **Structured Narrative of Functional Impact** table, expect to see functional data with each PubMed article separated in tabs if there are multiple articles
4. Check links and text formatting

### CODE REVIEW

File: src/clincoded/static/components/variant_central/index.js [hunk 1]
```diff
@@ -328,6 +337,150 @@ var VariantCurationHub = createReactClass({
...
+                this.setState({ loading_ldhFuncData: false });
+            }
+        };
+
+        // Sets pubmed articles in the @afisObject and replaces the AFIS with the completed @afisObject
+        const setPubmedArticles = (afisObject, articles) => {
+            if (!Array.isArray(articles)) {
+                return;
+            }
+            articles.forEach((article, articleIdx) => {
```
Comment (markmandell) [inline src/clincoded/static/components/variant_central/index.js]: Quick question! 

Is `articleIdx` meant to just be `articleId`? Or does it stand for " article index".

File: src/clincoded/ldh.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+@view_config(route_name='ldh', request_method='GET')
+def get_ldh_data(request):
+    try:
+        variant_id = request.matchdict['variant_id']
+    except (ValueError, TypeError, KeyError) as e:
+        return http_error(HTTPBadRequest(), request)
+    try:
+        ldh_url = 'https://ldh.clinicalgenome.org/ldh/Variant/id/' + variant_id
+        ldh_data = requests.get(ldh_url)
```
Comment (bryanwulf) [inline src/clincoded/ldh.py]:
```suggestion
        ldh_data = requests.get(ldh_url, timeout=10)
```

File: src/clincoded/ldh.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+@view_config(route_name='afis', request_method='GET')
+def get_allele_frequency_impact_statements(request):
+    try:
+        afis_id = request.matchdict['afis_id']
+    except (ValueError, TypeError, KeyError) as e:
+        return http_error(HTTPBadRequest(), request)
+    try:
+        afis_url = 'https://ldh.clinicalgenome.org/fdr/AlleleFunctionalImpactStatement/id/' + afis_id
+        afis_record = requests.get(afis_url)
```
Comment (bryanwulf) [inline src/clincoded/ldh.py]:
```suggestion
        afis_record = requests.get(afis_url, timeout=10)
```

File: src/clincoded/static/components/variant_central/interpretation/functional.js [hunk 4]
```diff
@@ -53,11 +62,97 @@ var CurationInterpretationFunctional = module.exports.CurationInterpretationFunc
...
+            }
+            const afisKeys = newAfis && Object.keys(newAfis);
+            return afisKeys.some(key => {
+                const oldStatements = oldAfis[key] && oldAfis[key].statements;
+                const newStatements = newAfis[key] && newAfis[key].statements;
+                if (!oldStatements || !newStatements || oldStatements.length !== newStatements.length) {
+                    return true;
+                }
+                return newStatements.some((statement, index) => {
+                    if (statement.rev !== oldStatements[index].rev) {
```
Comment (bryanwulf) [inline src/clincoded/static/components/variant_central/interpretation/functional.js]: May want to check that ```oldStatements[index]``` is defined before attempting to access ```oldStatements[index].rev```.

File: src/clincoded/static/components/variant_central/interpretation/functional.js [hunk 5]
```diff
@@ -53,11 +62,97 @@ var CurationInterpretationFunctional = module.exports.CurationInterpretationFunc
...
+                        return true;
+                    }
+                });
+            });
+        };
+        savedEvals.some(evaluation => {
+            if (['BS3', 'PS3'].indexOf(evaluation.criteria) > -1) {
+                const funcDataObjDiffFlag = isDiffAfisVersion(newFuncData, evaluation.functional.functionalData);
```
Comment (bryanwulf) [inline src/clincoded/static/components/variant_central/interpretation/functional.js]:
```suggestion
                const funcDataObjDiffFlag = isDiffAfisVersion(newFuncData, evaluation.functional && evaluation.functional.functionalData);
```
Not sure that ```evaluation.functional``` is certain to exist...

**Pull request merged**